Move closer to a series-centric website

yrodiere opened the pull request:
This PR changes a few things so that we can display more series-centric information, i.e. information related to a specific release series (5.0, 5.2, etc.).

I pushed the changes to staging.

## Main changes for visitors

 * The "Downloads" menu entry is replaced by "Releases" menu entry
 * "Releases" page:
   * The "Releases" page now puts more emphasis on the series (e.g. "5.2") and less on the release itself (e.g. "5.2.11").
   * The "Releases" page now includes links to the documentation for each series
   * The "Releases" page now includes links to a dedicated page for each series
   * The "Releases" page now includes a compatibility matrix, displaying for each series which versions of Java, JPA, Elasticsearch, etc. it's compatible with.
   * Example: http://staging.hibernate.org/search/releases/
 * Dedicated pages have been added for each series
   * They most notably include compatibilty information, links to the main documentation (user guide) and migration guide, a "What's new" section with information about the features of this series, and a list of all releases in this series with download links and links to the announcement.
   * Example: http://staging.hibernate.org/search/releases/5.8/
     * Note that in this example, we have a huge "What's new" section, but this is only because we messed up and delivered a huge version instead of multiple small ones. You'll probably want to keep the "What's new" section much smaller in other projects.

## Main changes for us

 * Each project now declares "integrations" in `site.yml`, providing a name and a URL. This is to be displayed in the compatibilty matrix. "integrations" can be whatever you want: Java itself, JPA, a supported DBMS, ...
 * YAML files for releases are now grouped in one folder for each series
 * some information in the YAML files for releases is now unnecessary (displayed, version, version_family)
 * each series folder contains a `series.yml` file with information about the series. This includes in particular:
   * "summary": a brief summary of the content of this series
   * "displayed": whether or not this series should be displayed
   * "integration_constraints": the constraints to be displayed in the compatibility matrix
 * Each "series" (i.e. minor version) must now have its own folder and related index.adoc file under "<project>/releases/". The file itself can be empty (except for Awestruct directives), but by adding content you can give more detailed information about the changes in this series.

## What you need to do

I took care of the groundwork, but there still is some work to do for each project before we can publish the new version of the website.

Absolutely necessary:

 * We must make sure to have exactly one YAML release file for each release in a displayed series. The main reason is that we generate a link to all JIRA tickets for a series, and if some releases are not present in YAML files, the list of tickets will be incomplete.
   * In ORM, the custom seems to be to remove release files when they are not the latest release for a given series. This should not happen anymore, we now need that data.
     I managed to retrieve most files from the git history, but some files were never there in the first place, for instance for all releases before 4.2.7. Someone will have to add them back, including the CR/Alpha/Betas.
   * In Validator, my understanding is that some versions were omitted on purpose, such as 6.0.0.Final. You must add YAML files for those versions regardless, but you can hide the versions on the site by adding `displayed: false` to the YAML file. 

Optional:

 * You may want to expand the compatibility matrices. For now I only added obvious information, such as "Java version" or "JPA version" for Hibernate ORM, but maybe you want to add compatibility information for each DBMS, or something else.
 * If you want, you can add more information about each series in the dedicated files: `<project>/releases/<series>/index.adoc`
 * For ORM, migration guides are currently on GitHub, while for Search they are on hibernate.org. I added support for linking to the migration guide of a given releases if it's on hibernate.org, but if the migration guides are on GitHub I can't easily check whether the migration guides exist. Maybe we'll want to move ORM migration guides back to hibernate.org... ?

yrodiere committed:
Add support for displaying information about release series

yrodiere committed:
Display release series for Hibernate Search

yrodiere committed:
Display release series for Hibernate OGM

yrodiere committed:
Display release series for Validator

yrodiere committed:
Restore ORM release files that have been deleted

yrodiere committed:
Display release series for Hibernate ORM

yrodiere committed:
Remove "Downloads" entries from the menu (we'll use "Relases")

yrodiere committed:
Move URL generation code to a helper

So that we don't need to copy/paste the code everywhere.

yrodiere committed:
Allow per-series documentation paths

emmanuelbernard commented:
@yrodiere I would write a doc for posterity based on your big message here. This is something people will have to refer to to understand how the website works.

emmanuelbernard commented:
Also, I think in the series page, there should be a prominent link or button to the actual specific serie page. Right now, you ahve a button for download and the doc and to find the series link, you need to click on the more obsucre "More on the 5.8 series "

What do others think?

yrodiere committed:
"More info" => "Release announcement"

yrodiere committed:
Fix type: "release-bottons" => "release-buttons"

yrodiere committed:
Remove ORM compatibilty info from the series list for Search

We already have that info in the compatibility matrix

yrodiere committed:
Move series-specific hero unit to a partial to avoid unnecessary vertical spaces

yrodiere committed:
Replace prominent download buttons with links to the series page

yrodiere committed:
Remove nonsensical uses of "row-fluid" or inexistant CSS classes

yrodiere committed:
Move the Red Hat advertisement out of the way in the "releases" pages

Because we seriously need to move the useful content higher on these
pages.

yrodiere committed:
Add download/maven instructions to the per-series page

yrodiere committed:
Display series in thumbnails in the "releases" page

yrodiere committed:
Add a link back from series-specific page to the "Releases" page

yrodiere committed:
Add series submenu entries under the "Releases" menu entry

yrodiere committed:
Define the title in <head> for series-specific pages

yrodiere committed:
Clean up Maven coordinates in project/release data

It should be a bit more obvious that:

 * there's an artifact ID in the site.yml that relates to the main
artifact of each project
 * there are groupId/artifactId *patterns* in the site.yml used for
links to Nexus search
 * group_id can be overridden by series, not only by release

yrodiere committed:
Document the latest changes in the survival guide

yrodiere committed:
Remove useless summaries from the release files

We used to put information about the current status of the series in
there, which lead to some summaries being repeated over and over. This
is no longer necessary, as we clearly explain what each series contains.
So if the release summary doesn't explain the content of this exact
release, it's useless.

yrodiere committed:
Add missing release files for displayed series

yrodiere committed:
Use a lighter styling in artifact lists (more margins, less font-weight)

yrodiere committed:
Do not highlight the "releases" menu entry when on a series-specific page
